### PR TITLE
Allow local Spacy Models to be loaded in NLP Engine

### PR DESCRIPTION
**Pull request by TSunny007**
Currently `SpacyNlpEngine` does not support use of local model paths. This is important for offline/airgapped environments where it is not possible to download spacy model on the fly

## Change Description

Follows Spacy's internal logic in `.load`
```py
if is_package(name):  # installed as package
    465     return load_model_from_package(name, **kwargs)  # type: ignore[arg-type]
    466 if Path(name).exists():  # path to model data directory
    467     return load_model_from_path(Path(name), **kwargs)  # type: ignore[arg-type]
```

## Issue reference

This PR fixes issue #XX

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [ ] I have signed the CLA (if required)
- [ ] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

File: presidio-analyzer/presidio_analyzer/nlp_engine/spacy_nlp_engine.py [hunk 1]
```diff
@@ -59,7 +60,7 @@ def load(self) -> None:
 
     @staticmethod
     def _download_spacy_model_if_needed(model_name: str) -> None:
-        if not spacy.util.is_package(model_name):
+        if not (spacy.util.is_package(model_name) or Path(name).exists()):
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/nlp_engine/spacy_nlp_engine.py:63]: Should it be `name` or `model_name`?

Comment (TSunny007) [reply presidio-analyzer/presidio_analyzer/nlp_engine/spacy_nlp_engine.py]: Ah sorry! I fat fingered this change. Fixed!

**Review by omri374 — APPROVED** [review]

**Comment by omri374** [comment]
Thanks!

**Pull request merged**